examples/rag/rag_agents.py

```diff
@@ -0,0 +1,189 @@
+# -*- coding: utf-8 -*-
+"""
+This example shows how to build an agent with RAG (backup by LlamaIndex)
+"""
+
+from typing import Optional
+from llama_index.core import SimpleDirectoryReader
+from langchain_community.document_loaders import DirectoryLoader
+
+from agentscope.prompt import PromptType
+from agentscope.agents.agent import AgentBase
+from agentscope.prompt import PromptEngine
+from agentscope.message import Msg
+from agentscope.models import load_model_by_config_name
+from agentscope.rag.llama_index_rag import LlamaIndexRAG
+from agentscope.rag.langchain_rag import LangChainRAG
+
+
+class RAGAgent(AgentBase):
```

Comment:
Maybe `RAGAgentBase` to remind users this is an abstract/base class, which cannot be used directly.

Reply:
updated